### PR TITLE
RHOAIENG-7489: Code-Server: Prevent auto updating of the pre-installed extensions

### PR DESCRIPTION
**Pull request by daniellutz**
This PR aims to disable the auto update from installed extensions in the Code Server notebook image.

**Jira:**
- [RHOAIENG-7489 - Code-Server: Prevent auto updating of the pre-installed extensions](https://issues.redhat.com/browse/RHOAIENG-7489)

**References:**
- [Visual Studio Code - Opt out of extension updates](https://code.visualstudio.com/docs/supporting/FAQ#_opt-out-of-extension-updates)
- [Visual Studio Code - Extension auto-update](https://code.visualstudio.com/docs/editor/extension-marketplace#_extension-autoupdate)

## Description
All extensions installed on Code Server will try to update automatically. To keep stability on the deployed notebook image, we aim to disable auto-update extensions, in a way that the user will update if it's really necessary.

Before this change, the extensions were marked for auto updates, as follows:

pre-autoupdate

After the code change, now the user can select to auto update extensions on their own, without the risk of having them updated in a not ideal timing:

post-autoupdate

## How Has This Been Tested?
This has been locally tested, which consisted on running the CodeServer image locally with the following commands (example of code-server on ubi9-python-3.9, but the same command was executed for python-3.11):

```bash
$ export QUAY_IO=quay.io/{user}/workbench-images
$ export WORKBENCH_RELEASE=2024b
$ make codeserver-ubi9-python-3.9 \
       -e IMAGE_REGISTRY=$QUAY_IO \
       -e RELEASE=$WORKBENCH_RELEASE \
       -e PUSH_IMAGES=no \
       -e CONTAINER_BUILD_CACHE_ARGS=""
```

This way the image could easily be built and tested, without pushing any images to our quay repository yet.

Now you can run with Podman:

```bash
$ export LATEST_TAG=`podman images --format "{{.Repository}}:{{.Tag}}" | grep "$QUAY_IO:codeserver-ubi9-python-3.9-$WORKBENCH_RELEASE" | sort -r | head -n1 | cut -d':' -f2`
$ podman run -it -p 8787:8787 $QUAY_IO:$LATEST_TAG
```

From there, you could open up in the browser, click on the `Extensions` tab on the left and check each extension.

## Merge criteria:
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by jiridanek** [comment]
It does what I was proposing back in the day in a comment on

* https://github.com/atheo89/notebooks/commit/0d1ab4d33feb187e747f626e2d18b881ddc0a606#diff-bf1e8d5482ec091a2c81bf7d4f573ea7a669d8d7d1d469201ec1e5d797522f1eR26

and from the evidence in the description it seems to be working. For me it's good to be merged.

/lgtm

**Review by jstourac — COMMENTED** [review]
/lgtm

It's good we are keeping the python 3.9 code in sync. Just be aware that this change won't land in 2024.1 unless also relevant 2024a branch is updated. In the tracker ticket it's not explicitly stated what all images versions we want to fix this to. I guess that it's 2024b only, but maybe we should consider to specify this explicitly for the future?

**Comment by jiridanek** [comment]
> It's good we are keeping the python 3.9 code in sync.

I take this as implementing things in main branch; since we have 3.9 and 3.11 vscode image there, then that's where the change should land; unless there are requirements to get this to some specific images release or rhoai release, then let's default to main and just do what's natural

**Comment by daniellutz** [comment]
@jiridanek @jstourac

If there were requirements to fix a specific version, I would follow it up.

Since there were no such requirements, I fixed both that are available on main branch.

File: codeserver/ubi9-python-3.11/run-code-server.sh [hunk 1]
```diff
@@ -46,7 +46,9 @@ universal_json_settings='{
   "python.defaultInterpreterPath": "/opt/app-root/bin/python3",
   "telemetry.telemetryLevel": "off",
   "telemetry.enableTelemetry": false,
-  "workbench.enableExperiments": false
+  "workbench.enableExperiments": false,
```
Comment (atheo89) [inline codeserver/ubi9-python-3.11/run-code-server.sh:49]: Thank you, Daniel, for opening this pull request.

I tested it using the generated image from the PR:
`quay.io/opendatahub/workbench-images:codeserver-ubi9-python-3.11-pr-739`

After comparing the old image with the one from this PR, I can confirm that the auto-update feature is now disabled by default. Users will need to manually enable it if desired.

Here’s a screenshot from the old image (Auto-update enabled by default):

image

And here’s the new image from this PR (Auto-update successfully disabled by default):

image

/lgtm
/approve

**Review by atheo89 — APPROVED** [review]
Thank you, Daniel, for opening this pull request.

I tested it using the generated image from the PR:
`quay.io/opendatahub/workbench-images:codeserver-ubi9-python-3.11-pr-739`

After comparing the old image with the one from this PR, I can confirm that the auto-update feature is now disabled by default. Users will need to manually enable it if desired.

Here’s a screenshot from the old image (Auto-update enabled by default):

image

And here’s the new image from this PR (Auto-update successfully disabled by default):

image

/lgtm
/approve

**Pull request merged**